src/main/java/wooteco/subway/line/dao/LineDaoJdbcTemplate.java

```diff
@@ -0,0 +1,79 @@
+package wooteco.subway.line.dao;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Repository;
+import wooteco.subway.line.Line;
+
+import javax.sql.DataSource;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Repository
+public class LineDaoJdbcTemplate implements LineDao {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final DataSource dataSource;
+
+    public LineDaoJdbcTemplate(JdbcTemplate jdbcTemplate, DataSource dataSource) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public Optional<Line> findLineByName(String name) {
+        final String sql = "SELECT * FROM line WHERE name = ?";
+        return jdbcTemplate.query(sql, lineRowMapper(), name)
+                .stream()
+                .findAny();
+    }
+
+    @Override
+    public Line save(Line line) {
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("name", line.getName());
+        parameters.put("color", line.getColor());
+
+        SimpleJdbcInsert jdbcInsert = new SimpleJdbcInsert(dataSource)
```

Comment:
```suggestion
        SimpleJdbcInsert jdbcInsert = new SimpleJdbcInsert(jdbcTemplate.getDataSource())
```

jdbcTemplate가 가지고 있는 data source를 넘겨주는게 더 깔끔하겠네요

Reply:
오!! jdbcTemplate 에 있는 dataSource를 가져올 수 있군요!!